lib/src/http/uncased.rs

```diff
@@ -55,6 +56,22 @@ impl UncasedStr {
     pub fn as_str(&self) -> &str {
         &self.0
     }
+
+    /// Converts a `Box<UncasedStr>` into a `Uncased` without copying or allocating.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use rocket::http::uncased::Uncased;
+    ///
+    /// let uncased = Uncased::new("Hello!");
+    /// let boxed = uncased.clone().into_boxed_uncased_str();
+    /// assert_eq!(boxed.into_uncased(), uncased);
+    /// ```
+    pub fn into_uncased(self: Box<UncasedStr>) -> Uncased<'static> {
+        let boxed_str: Box<str> = unsafe { transmute(self) };
```

Comment:
I think we can do this without using `transmute` for a little more type-safety. Maybe something like `Box::from_raw(Box::into_raw(self) as *mut str)`? May need a `&*` at the front.